src/LINEBot/SignatureValidator.php

```diff
@@ -38,7 +38,7 @@ class SignatureValidator
      */
     public static function validateSignature($body, $channelSecret, $signature)
     {
-        if (!isset($signature)) {
+        if (trim($signature) === '') {
```

Comment:
This method may be called by developers.
https://github.com/line/line-bot-sdk-php/blob/master/src/LINEBot.php#L337-L340

How did you detect here is always not null?

Reply:
PHPStan consider `$signature` should never be `NULL` (Unless user is using it wrong of course), so it will mark this as violation.

`trim()` on `NULL` returns empty string `''` so that

```php
trim(NULL) === ''
```
will always returns true across all PHP versions, it checks not only `NULL` or empty strings but also blank strings. So it does not lossen the current checks, only stricten it to disallow blank string as `$signature` as well.

Also, this modification can makes PHPStan happy.

Reply:
Thanks.
We check variable is null with using isset.
We will use trim instead of isset after this p-r.